apps/api-example/src/app/slack-mail-provider.ts

```diff
@@ -44,4 +47,19 @@ export class SlackMailProvider extends BaseMailProvider {
       body: JSON.stringify(message)
     })
   }
+
+  async getTemplates(): Promise<MailProviderTemplate[] | MailProviderError> {
+    return [
+      {
+        name: 'SlackEmptyTemplate',
+        uniqueIdentifier: 'slack-empty',
+        createdAt: new Date(),
+        updatedAt: new Date()
+      }
+    ]
+  }
+
+  getTemplateUrl(template: MailTemplate): string {
```

Comment:
I prefer to leave it in the interface, since all other implementations need it. Only the Slack provider is a special case (it has no way to even define templates, and therefore no URLs either). Removing the method from the interface would mean that we have to check for its presence for all other providers (and remember implementing it if we ever add new providers).